controllers/database_controller.go

```diff
@@ -188,25 +188,55 @@ func (r *DatabaseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			// when database creation failed, don't requeue request. to prevent exceeding api limit (ex: against google api)
 			return r.manageError(ctx, dbcr, err, false)
 		}
+		ownership := []metav1.OwnerReference{
+			{
+				APIVersion: dbcr.APIVersion,
+				Kind:       dbcr.Kind,
+				Name:       dbcr.Name,
+				UID:        dbcr.GetUID(),
+			},
+		}
+
 		dbcr.Status.Phase = dbPhaseInstanceAccessSecret
-		err = r.createInstanceAccessSecret(ctx, dbcr)
+		// TODO: Return instanceSecret and set ownership
+
+		instanceSecret, err := r.createInstanceAccessSecret(ctx, dbcr)
 		if err != nil {
 			return r.manageError(ctx, dbcr, err, true)
+		} else if instanceSecret != nil && dbcr.Spec.Cleanup {
+			instanceSecret.SetOwnerReferences(ownership)
+			if err := r.Update(ctx, instanceSecret); err != nil {
+				return r.manageError(ctx, dbcr, err, false)
+			}
 		}
 		dbcr.Status.Phase = dbPhaseProxy
 		err = r.createProxy(ctx, dbcr)
 		if err != nil {
 			return r.manageError(ctx, dbcr, err, true)
 		}
 		dbcr.Status.Phase = dbPhaseSecretsTemplating
-		err = r.createTemplatedSecrets(ctx, dbcr)
+		// TODO: Return secret and set ownership
+		dbSecret, err := r.createTemplatedSecrets(ctx, dbcr)
 		if err != nil {
 			return r.manageError(ctx, dbcr, err, true)
+		} else if dbSecret != nil && dbcr.Spec.Cleanup {
+			logrus.Infof("Setting up OwnerReference %v", ownership)
+			dbSecret.SetOwnerReferences(ownership)
```

Comment:
why not define the ownership at creation in the related function?

Reply:
I've tried to add it to SecretBuilder and ConfigmapBuilder, but they are called from places where there is no context that is required for creating an owner reference. I could add it to the main reconcile function, pass it through a big chain of functions, add conditions everywhere, or just put it here. It may be not the best way of doing this, but otherwise, a lot of code should be restructured, and I'm not sure if it should be done within this PR.

For example, `fn SecretBuilder()` is called from every function that modifies `templateSecrets` and `connectionString` that are called from the main templateSecrets creator function that is moved to `database_helper.go`. So it was not really obvious where to put the ownership stuff.

Reply:
Would mostly supply the `ownership` as argument to the function like `createInstanceAccessSecret`, and apply there the `SetOwnerReferences` function.

and for the ownership variable definition at 191, think you could only make there the if condition if the ownership is an empty struct or filled with the reference to the db-operator.

But my assumption is also, that an empty ownership struct, should not really do anything at `SetOwnerReferences`.  But maybe I'm wrong.

Another question for me, issue is only about secrets and cm. But we also create the proxy deployment and cronjob. Should we also include this here, or a separate issue.

Reply:
Sounds good, I didn't think about that. I'll try.

I think other resources should be removed too, but I just forgot about them. So I'll do that